src/proto/topic.rs

```diff
@@ -227,9 +227,11 @@ impl<PI, R> State<PI, R> {
 impl<PI: PeerIdentity, R: Rng> State<PI, R> {
     /// Initialize the local state with a custom random number generator.
     pub fn with_rng(me: PI, me_data: Option<PeerData>, config: Config, rng: R) -> Self {
+        let max_payload_size =
+            config.max_message_size - super::Message::<PI>::postcard_header_size();
```

Comment:
a faulty configuration will still work in release mod with gross `max_payload_size` due to overflowing arithmetic. However unlikely, this should use checked arithmetic and use the default if it fails (or something else). Never underestimate stuff people do?

Reply:
Good point. I'm a bit undecided what to do exactly. I don't like the idea of a silent fallback to the default too much. I think I'd prefer to define a  `MIN_MAX_MESSAGE_SIZE` with some reasonable number (eg 512), and assert! that the max_message_size is below that, and document that `State::new` and `Topic::new` will panic if max_message_size is below this lower limit. What do you think?

Reply:
sounds good :+1:

Reply:
Implemented that.